### ISSUE TITLE
Pressing return in a list with text after the cursor doesn't create a new list item

### ISSUE THREAD

**Issue by marshallpierce** [HappenApps/Quiver#301]
```
• Foo<cursor>Bar
```

press enter results in

```
• Foo
Bar
```

but should be

```
• Foo
• Bar
```

Once the editor is in this state, it's difficult to recover. You need to cut or otherwise remove all text after teh cursor, press enter to get another bullet, and then add it in.

**Comment by ylian** [comment]
Is this in a text cell? I can't seem to reproduce it.

**Comment by marshallpierce** [comment]
Yes, in a text cell. I reproduced it while pasting things in, rather than typing directly. I've switched to Markdown cells just to escape this bug, but I'll fiddle with it a bit more to see if I can get a concrete repro.

**Comment by marshallpierce** [comment]
Ah yes, it's definitely to do with pasting. I'm pasting simple text strings from IntelliJ.

```
• Foo
• <pasted content><cursor>
```

hit enter

```
• Foo
• <pasted content>
<cursor>
```

and then the list is horked. If I paste as "match style" then it doesn't break, but for some reason regular paste from IntelliJ appears to be interpreted as though it had line breaks coming with it or something.

**Comment by ylian** [comment]
Thanks. It seems that the pasted string has some built-in styles applied. If you can export the note with the borked list and send to me, I can take a look. My email is ylian@happenapps.com

**Comment by marshallpierce** [comment]
Emailed, thanks.

**Comment by ylian** [comment]
Got it, thanks.

Indeed the pasted text has inline styles, actually in this case it's set as `<pre>...</pre>` to preserve syntax highlighting from IntelliJ.

![screen shot 2015-12-10 at 2 37 25 pm](https://cloud.githubusercontent.com/assets/984895/11726247/c095e4d4-9f4b-11e5-9803-38d5f6d137f7.png)

That will certainly break the list. Closing this since this is caused by the inline styles of the pasted text.

**Comment by marshallpierce** [comment]
Uh, so "just don't paste from intellij" into a list? That's not OK for an editor aimed at programmers.

**Comment by ylian** [comment]
I mean you should paste as plain text ("Paste and Match Style").

**Comment by marshallpierce** [comment]
This doesn't break, say, Evernote...

**Comment by ylian** [comment]
You're right. Evernote seems to do something behind the scenes. I will reverse engineer their JavaScript and figure it out. Reopen.
